### PR TITLE
Support customized path for war deployment

### PR DESCRIPTION
**Pull request by Flanker32**
Support customized path for war deployment

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java [hunk 1]
```diff
@@ -88,7 +89,13 @@ public void deploy(File file) {
     }
 
     public void deploy(DeployType deployType, File target) {
-        getWebAppInner().deploy(com.azure.resourcemanager.appservice.models.DeployType.fromString(deployType.getValue()), target);
+        deploy(deployType, target, null);
+    }
+
+    @Override
+    public void deploy(DeployType deployType, File target, String targetPath) {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java:96]: targetFile

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java [hunk 2]
```diff
@@ -88,7 +89,13 @@ public void deploy(File file) {
     }
 
     public void deploy(DeployType deployType, File target) {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebApp.java:91]: targetFile

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java [hunk 3]
```diff
@@ -209,27 +207,32 @@ private IWebAppDeploymentSlot updateDeploymentSlot(final IWebAppDeploymentSlot s
 
     private void deployArtifacts(IAppService target, WebAppConfig config) throws AzureExecutionException {
         // This is the codes for one deploy API, for current release, will replace it with zip all files and deploy with zip deploy
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java:209]: rename the var name accordingly

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java [hunk 4]
```diff
@@ -102,6 +108,19 @@ protected void validate(Supplier<String> validator) throws AzureExecutionExcepti
         }
     }
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java:110]: why you filter isExternalResource in convertResourcesToArtifact, it is not expected according to the method name.

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java [hunk 5]
```diff
@@ -102,6 +108,19 @@ protected void validate(Supplier<String> validator) throws AzureExecutionExcepti
...
 
+    protected List<MavenArtifact> convertResourcesToArtifact(List<Resource> resources) {
```
Comment (andxu) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java:111]: I think it should be static

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java (hunk 4 above)
Comment (Flanker32) [reply azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/AbstractConfigParser.java:110]: Renamed to `parseArtifactsFromResources`

**Review by andxu — APPROVED** [review]

**Pull request merged**